app/controllers/concerns/client_sortable.rb

```diff
@@ -24,38 +24,52 @@ def filtered_and_sorted_clients(default_order: nil)
 
   # see if there are any overlapping keys in the provided params and search/sort set
   def has_search_and_sort_params?
-    (params.keys.map(&:to_sym) & search_and_sort_params).any?
+    overlapping_keys = (params.keys.map(&:to_sym) & search_and_sort_params)
+    hash_params = params.try(:to_unsafe_h) || params
+    overlapping_keys.any? && hash_params.slice(*overlapping_keys).any? { |_, v| v.present? }
   end
 
   private
 
   def setup_sortable_client
-    reset_filter_params if params[:clear]
+    delete_cookie if params[:clear]
     @sort_column = clients_sort_column
     @sort_order = clients_sort_order
-    @filters = {
-      search: normalize_phone_number_if_present(params[:search]),
-      status: status_filter,
-      stage: stage_filter,
-      assigned_to_me: params[:assigned_to_me],
-      unassigned: params[:unassigned],
-      needs_response: params[:needs_response],
-      year: params[:year],
-      vita_partner_id: params[:vita_partner_id].present? ? params[:vita_partner_id].to_i : nil,
-      assigned_user_id: params[:assigned_user_id].present? ? params[:assigned_user_id].to_i : nil,
+    filter_source = has_search_and_sort_params? ? params : cookie_filters
+    @filters = filters_from(filter_source)
+    set_cookie
+  end
+
+  def delete_cookie
+    cookies.delete(filter_cookie_name) if filter_cookie_name.present?
+  end
 
+  def set_cookie
```

Comment:
If possible, I'd like to see a unit spec that covers this since it's somewhat complicated. I see it's also covered in the feature spec.